var/spack/repos/builtin/packages/ucx/package.py

```diff
@@ -60,6 +60,24 @@ class Ucx(AutotoolsPackage, CudaPackage):
             description='Enable KNEM support')
     variant('xpmem', default=False,
             description='Enable XPMEM support')
+    variant('cma', default=False,
+            description="nable Cross Memory Attach")
+    variant('avx', default=False,
+            description="Enable AVX option")
```

Comment:
Can we avoid a variant and inspect the target instead to make a decision?

Reply:
Unfortunately there is no such generic option available with UCX. There are total five arch related flags, out of these compiler specific flags two flags --with-mcpu and --with-march searches for ThunderX architectures. Two other flags related to SSE. And the fifth flag we have added i.e --with-avx which suits best for latest architectures. 

If you want, we will remove --with-avx flag and update the package. Please suggest how to go about this.

Reply:
To be clear, for the `avx` variant what I mean is that you can automatically do something like:
```python
if 'avx' in self.spec.target:
    # Do something based on avx detection
```
and drop the variant. This is in line with what we do in other packages, like `gromacs`.

Reply:
Addressed the same. Thanks.